app/src/test/java/org/fossasia/openevent/app/core/viewmodel/LoginViewModelTest.java

```diff
@@ -0,0 +1,128 @@
+package org.fossasia.openevent.app.core.viewmodel;
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule;
+import android.arch.lifecycle.Observer;
+
+import org.fossasia.openevent.app.common.Constants;
+import org.fossasia.openevent.app.core.auth.login.LoginViewModel;
+import org.fossasia.openevent.app.data.Preferences;
+import org.fossasia.openevent.app.data.auth.AuthService;
+import org.fossasia.openevent.app.data.network.HostSelectionInterceptor;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+@RunWith(JUnit4.class)
+public class LoginViewModelTest {
+
+    @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+    @Rule public TestRule rule = new InstantTaskExecutorRule();
+
+    @Mock private Preferences sharedPreferenceModel;
+    @Mock private AuthService authModel;
+    private final HostSelectionInterceptor interceptor = new HostSelectionInterceptor();
+
+    private static final String EMAIL = "test";
+    private static final String PASSWORD = "test";
+    @InjectMocks private LoginViewModel loginViewModel;
+    private static final Set<String> SAVED_EMAILS = new HashSet<>(Arrays.asList("email1", "email2", "email3"));
+
+    @Mock Observer<Boolean> login;
+    @Mock Observer<String> error;
+    @Mock Observer<Set<String>> email;
+    @Mock Observer<Boolean> progress;
+
+    @Before
+    public void setUp() {
+        loginViewModel = new LoginViewModel(authModel, interceptor, sharedPreferenceModel);
+        loginViewModel.getLogin().setEmail(EMAIL);
+        loginViewModel.getLogin().setPassword(PASSWORD);
+        initMocks(this);
+    }
+
+    @Test
+    public void shouldLoginSuccessfully() {
+        loginViewModel.getProgress().observeForever(progress);
+        loginViewModel.getLoginStatus().observeForever(login);
+
+        login.onChanged(false);
+        verify(login).onChanged(false);
+
+        progress.onChanged(true);
+        verify(progress).onChanged(true);
+
+        login.onChanged(true);
+        verify(progress).onChanged(true);
+
+        progress.onChanged(false);
+        verify(progress).onChanged(false);
+    }
```

Comment:
This test is not testing the ViewModel

This is testing itself

Call hello on speaker
Verify that hello was called on speaker